server/src/test/java/org/jetbrains/bsp/bazel/server/sync/languages/LanguagePluginServiceTest.java

```diff
@@ -0,0 +1,417 @@
+    }
+
+    @Test
+    public void shouldGetPluginForScala() {
+      // given
+      Set<Language> languages = HashSet.of(Language.SCALA);
+
+      // test
+      var plugin = languagePluginsService.getPlugin(languages);
+
+      // assert
+      assertEquals(ScalaLanguagePlugin.class, plugin.getClass());
+    }
+
+    @Test
+    public void shouldGetPluginForEmpty() {
+      // given
+      Set<Language> languages = HashSet.of();
+
+      // test
+      var plugin = languagePluginsService.getPlugin(languages);
+
+      // assert
+      assertEquals(EmptyLanguagePlugin.class, plugin.getClass());
+    }
+
+    @Test
+    public void shouldGetPluginForThrift() {
+      // given
+      Set<Language> languages = HashSet.of(Language.THRIFT);
+
+      // test
+      var plugin = languagePluginsService.getPlugin(languages);
+
+      // assert
+      assertEquals(ThriftLanguagePlugin.class, plugin.getClass());
+    }
+  }
+
+  @Nested
+  @DisplayName("Tests for the method shouldGetSourceSet")
+  class shouldGetSourceSetTest {
+
+    LanguagePluginsService languagePluginsService;
+    Path tmpRepo;
+
+    @BeforeEach
+    void beforeEach() {
+      var bazelInfo = new BasicBazelInfo("", Paths.get(""));
+      var bazelPathsResolver = new BazelPathsResolver(bazelInfo);
+      var jdkResolver = new JdkResolver(bazelPathsResolver, new JdkVersionResolver());
+      var javaLanguagePlugin = new JavaLanguagePlugin(bazelPathsResolver, jdkResolver, bazelInfo);
+      var scalaLanguagePlugin = new ScalaLanguagePlugin(javaLanguagePlugin, bazelPathsResolver);
+      var cppLanguagePlugin = new CppLanguagePlugin();
+      var thriftLanguagePlugin = new ThriftLanguagePlugin(bazelPathsResolver);
+      languagePluginsService =
+          new LanguagePluginsService(
+              scalaLanguagePlugin, javaLanguagePlugin, cppLanguagePlugin, thriftLanguagePlugin);
+    }
+
+    private Path createFileAndWrite(String dirString, String filename, String content) {
+      try {
+        tmpRepo = Files.createTempDirectory("");
+        Path dirPath = tmpRepo.resolve(dirString);
+        Files.createDirectories(dirPath);
+        Path filePath = dirPath.resolve(filename);
+        Files.createFile(filePath);
+
+        FileOutputStream outputStream = new FileOutputStream(filePath.toString());
+        byte[] strToBytes = content.getBytes();
+        outputStream.write(strToBytes);
+        outputStream.close();
+
+        return filePath;
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
+      return null;
+    }
+
+    @AfterEach
+    void afterEach() {
+      File tmpDir = tmpRepo.toFile();
+      deleteRecursively(tmpDir);
+    }
+
+    private boolean deleteRecursively(File dir) {
+      File[] files = dir.listFiles();
+      if (files != null) {
+        for (File file: files) {
+          deleteRecursively(file);
+        }
+      }
+      return dir.delete();
+    }
+
+    @Test
+    public void shouldGetSourceSetForJavaFile() {
```

Comment:
it'd be great to add a test case with invalid file syntax - for example two imports in java or something like that